### DISCUSSION TITLE
Get current active widget information/ID

### DISCUSSION THREAD

**Discussion by najaarv** [eclipse-theia/theia#8409] [General] [answered]
Hi, I have multiple widgets visible by default all over the layout. 
1) How can I get the currently active widget(at least in the side panels) information or it's ID?
2) How can I get the expanded state of a widget present in a side panel?
TIA.

**Comment by vince-fugnitto** [comment, marked as answer]
@najaarv you can use the `ApplicationShell` to get the information:
- hhttps://eclipse-theia.github.io/theia/docs/next/classes/core.applicationshell-2.html#activewidget

Please review the [api documentation](https://eclipse-theia.github.io/theia/docs/next/index.html) which is also referenced in the readme.
For coding examples, you can also look at the sources, for example (the `console`):

https://github.com/eclipse-theia/theia/blob/9d2dcd0a0923d7f7d8def0b7377fc9667706c967/packages/console/src/browser/console-manager.ts#L27-L35
